@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * The object of the event, fired when the
+ * a chunk of data uploaded to the storage.
+ */
+export interface UploadChunkEvent {
+    size: number;
+    number: number;
+    partId: string;
+}
+
+/**
+ * The class for uploading data to the storage
+ * by chunks.
+ *
+ * You can use this class in the browsers and
+ * Node.js for uploading large files.
+ *
+ * The file does not load in the RAM. Instead of
+ * loading the whole file of RAM, the file reads
+ * and uploads in chunks.
+ * So you can upload really large files.
+ *
+ * In browser you can create the instance with
+ * `File` or `Blob` object or buffer of the uploaded file.
+ *
+ * In Node.js you can create the instance with a path to the
+ * file or with a buffer of the data.
+ *
+ * Also, you can create your own implementation of reading data
+ * from some your place.
+ * Just implement the Data class and set it.
+ */
+export class MultiPartUploadWrapper {
+    constructor(opts: {
+        /**
+         * The data to upload.
+         * Use in brorsers: File | `Blob` | `ArrayBufferLike` | `Data`.
+         * Use in Node.js: string (path to file) | ArrayBufferLike | [[Data]].
+         */
+        data: string | File | Blob | ArrayBufferLike | Data;
+        /**
+         * The [[OlpClientSettings]] instance.
+         */
+        settings: OlpClientSettings;
+    }) {}
+
+    /**
+     * The method to upload data by chunks to the
+     * `Blob` storage.
+     *
+     * The method starts the `MultiPart Upload` process
+     * to the Blob V1 or V2, then uploads the data in chunks,
+     * and then complete the `MultiPart Upload` process
+     */
+    async upload(opts: {
+        /**
+         * The hrn string of the catalog to upload.
+         */
+        catalogHrn: string;
+
+        /**
+         * The ID of the Layer to upload.
+         */
+        layerId: string;
+
+        /**
+         * The version of the Blob storage.
+         * Supporting 1 or 2.
+         */
+        blobVersion: number;
+
+        /**
+         * The key of uploading data to use.
+         * The key means the datahandle if you are using
+         * Blob V1.
+         */
+        handle: string;
+
+        /**
+         * Content type of the uploaded object.
+         * Must be equal to the layer content type.
+         *
+         */
+        contentType: string;
+
+        /**
+         * The parallel requests for uploading chunks.
+         * Default: 6.
+         * @optional
+         */
+        parallelRequests?: number;
+
+        /**
+         * The size of each chunk.
+         * Minimum 5M, maximum 5GB.
+         * The default is 5MB.
+         * The default chunk size will be used in case,
+         * you set the out-of-range size.
+         * @optional
+         */
+        chunkSizeMB?: number;
+
+        /**
+         * Content encoding of the uploaded object.
+         * Must be equal to the layer content encoding.
+         * @optional
```

Comment:
Here and below, delete "@optional". TypeDoc will mark this method as optional because of the "?" sign.